lib/OpenQA/Command.pm

```diff
@@ -122,8 +122,9 @@ sub retry_tx ($self, $client, $tx, $retries = undef, $delay = undef) {
     $client->connect_timeout($ENV{MOJO_CONNECT_TIMEOUT} // 30);
     $delay //= $ENV{OPENQA_CLI_RETRY_SLEEP_TIME_S} // 3;
     $retries //= $ENV{OPENQA_CLI_RETRIES} // 0;
+    my $factor = $ENV{OPENQA_CLI_RETRY_FACTOR} // 1;
     my $start = time;
-    for (;; --$retries) {
+    do {
```

Comment:
I'd prefer to keep the for-loop. It just made more sense. We do not need to do any checking in the loop-statement itself because our exit-condition is the ` && $retries > 0`-part in the code below which probably should stay (and also stays after your commit). Making this a while-loop with exit-condition-check means we now have two exit-conditions which makes the code harder to understand overall.

Reply:
You could make it a `for (;;) { … }`-loop and change ` && $retries > 0` to ` && $retries-- > 0` if you want to keep the decrement and condition in one place. This would possibly be the cleanest solution and wouldn't introduce another exit-condition. However, changing how this loop works is probably out of scope for your commit anyway so I suggest to simply take that change back and really only add the exponential backoff in your commit.

Reply:
you are right in the sense that two exit conditions are present. But i see more logic to just remove the `&& $retries > `. do..while ensures blocks runs at least one and keep iterate for all the retries. WDYT?

Reply:
I would prefer to leave the original loop logic untouched at this point.

Reply:
seems there is a conflict of opinions and I am not convinced. I encouraged to go on with this after an offline comment while we examined some problem. And from my POV `for` implies a certain number of iteration, while the `do--while` says run at least one and then try again under true condition. it is more clear. What;s the actual reason to keep the `for` other than personal preference?

Reply:
@b10n1k I would prefer if you can please speed this up just getting this over with. This trivial change is now taking quite long and becoming costly. There should not be unrelated changes in pull requests just "for the sake of it". If you like to introduce a do-while then do that in a separate pull request with an according explicit explanation

Reply:
> What;s the actual reason to keep the for other than personal preference?

I have stated that before:

> Making this a while-loop with exit-condition-check means we now have two exit-conditions which makes the code harder to understand overall.

So I just want to keep the number of conditions for this loop's control flow at one and your change increases them to two.

Reply:
yea but that means that i have missed to remove the `&& $retries > 0`. it doesnt explain why to switch back to the `for`. I hope my concern is clear.

Reply:
I understand that

    for (;; --$retries) {
    ...
    }

does what we want and is easier to read.

Or you think there is a bug in the existing code?

Reply:
https://github.com/os-autoinst/openQA/pull/5506#discussion_r1530148325

removing the `&& $retries > 0` you stay with a clear block which says run once and then repeat while condition is satisfied. I dont think is any bug. it is how you read it. Also i dont remember the argument from the one who suggest that we shouldnt use `for` here, so i cant comment on that

Reply:
Lets forget it. I pushed a changed and as @perlpunk mentioned is not good. I reverted and put back the `for` loop